onnxruntime/core/session/invoker_abi_impl.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include <string>
+#include <iostream>
+
+#include "core/framework/error_code_helper.h"
+#include "core/framework/ort_value.h"
+#include "core/framework/execution_provider.h"
+#include "core/session/onnxruntime_c_api.h"
+#include "core/session/ort_apis.h"
+#include "core/eager/ort_kernel_invoker.h"
+#include "abi_session_options_impl.h"
+#include "core/session/ort_env.h"
+#include "core/graph/basic_types.h"
+
+namespace onnxruntime {
+struct OrtInvokerImpl {
+  IOnnxRuntimeOpSchemaRegistryList custom_schemas;
+  std::unique_ptr<onnxruntime::ORTInvoker> invoker;
+};
+}  // namespace onnxruntime
+
+ORT_API_STATUS_IMPL(OrtApis::CreateInvoker,
+                    _In_ const OrtEnv* env_,
+                    _In_ const OrtSessionOptions* options,
+                    size_t provider_index,
+                    _Outptr_ OrtInvoker** invoker_) {
+  API_IMPL_BEGIN
+  // Create the logger
+  const onnxruntime::logging::Logger& logger = env_->GetLoggingManager()->DefaultLogger();
+
+  // Create the provider
+  ORT_ENFORCE(provider_index < options->provider_factories.size(), "provider_index (" + std::to_string(provider_index) + ") must be less than the provider list size (" + std::to_string(options->provider_factories.size()) + ").");
+  std::shared_ptr<onnxruntime::IExecutionProvider> provider = options->provider_factories.at(provider_index)->CreateProvider();
+
```

Comment:
need to be a little bit careful about the EP lifetime.
In the training/eager mode build, the EP instances are managed in training module and shared across multiple session / invokers. but here, it seems the invoker instance own the EP lifetime. Who will manage the invoker instance? if it is managed by user code, how will we make it work with the EP instances within training module? 
for many hardware, it might not work well if multiple EP instances created on the same device.